Context/Adapter/ContextAdapterInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Context\Adapter;
+
+use FOS\RestBundle\Context\ContextInterface;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+interface ContextAdapterInterface
+{
+    /**
+      * Specialize a standard context.
+      *
+      * @param ContextInterface $context
+      * @param int            $direction
+      * @param mixed            $serializer
+      *
+      * @return mixed
+      *
+      * @throws \LogicException if the specialization is not supported
+      */
+     public function specialize(ContextInterface $context, $direction, $serializer);
```

Comment:
I didn't set the serializer on the adapter because ``ViewHandler::getSerializer`` allows to change dynamically the serializer by extending the class, no ?